pyro/contrib/gp/__init__.py

```diff
@@ -1,15 +1,136 @@
 from __future__ import absolute_import, division, print_function
 
+from torch.autograd import Variable
 import torch.nn as nn
-from torch.nn import Parameter
+from torch.distributions import transform_to
 
+import pyro
+import pyro.distributions as dist
 
-class InducingPoints(nn.Module):
 
-    def __init__(self, Xu, name="inducing_points"):
-        super(InducingPoints, self).__init__()
-        self.inducing_points = Parameter(Xu)
-        self.name = name
+class Parameterized(nn.Module):
+    """
+    Parameterized class.
 
-    def forward(self):
-        return self.inducing_points
+    This is a base class for other classes in Gaussian Process.
+    By default, a parameter will be a ``torch.nn.Parameter`` containing ``torch.FloatTensor``.
+    To cast them to the correct data type or GPU device, we can call methods such as
+    ``.double()``, ``.cuda(device=0)``,...
+    See `torch.nn.Module
+    <http://pytorch.org/docs/master/nn.html#torch.nn.Module>`_ for more information.
+    """
+
+    def __init__(self):
```

Comment:
nit: What do you think of adding `name` as an `__init__` arg rather than setting `self._name = ...` in derived classes? This seems a little safer to me.